### PR TITLE
Basic tests for serializers

### PR DESCRIPTION
**Pull request by MarcosFierro**

### CODE REVIEW

File: db/migrate/20191007205911_create_comments.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class CreateComments < ActiveRecord::Migration[5.2]
   def change
     create_table :comments do |t|
-      t.integer :user_id, :null => false
-      t.integer :article_id, :null => false
+      t.bigint :user_id, :null => false
+      t.bigint :article_id, :null => false
```
Comment (AlvaroSenorale) [inline db/migrate/20191007205911_create_comments.rb]: you should use `reference`, or at least add an index

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -41,7 +41,7 @@
   config.use_transactional_fixtures = true
 
   config.before(:suite) do
-    DatabaseCleaner.clean_with(:truncation)
+    DatabaseCleaner.clean_with(:truncation, except: %w(ar_internal_metadata))
```
Comment (AlvaroSenorale) [inline spec/rails_helper.rb]: why did you need to add this?

File: spec/serializers/article_serializer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+describe ArticleSerializer do
+  let!(:user) { create(:user) }
+  let(:article) { create(:article) }
+
+  let(:hash) { {:data=>{:id=>"#{article.id}", :type=>:article, :attributes=>{:id=>article.id, :title=>article.title, :description=>article.description}, :relationships=>{:comments=>{:data=>[]}, :user=>{:data=>{:id=>"1", :type=>:user}}}}} }
```
Comment (AlvaroSenorale) [inline spec/serializers/article_serializer_spec.rb]: I think you should run `rubocop`, this hash could be cleaner

File: spec/serializers/article_serializer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  let(:json) { {"data":{"id":"#{article.id}","type":"article","attributes":{"id":article.id,"title":article.title,"description":article.description},"relationships":{"comments":{"data":[]},"user":{"data":{"id":"1","type":"user"}}}}} }
```
Comment (AlvaroSenorale) [inline spec/serializers/article_serializer_spec.rb]: this one too

File: spec/serializers/article_serializer_spec.rb (hunk 3 above)
Comment (AlvaroSenorale) [reply spec/serializers/article_serializer_spec.rb]: Also I would called like `article_attributes_hash`, and `article_attributes_json`. I mean, in a more descriptive way

File: spec/serializers/comment_serializer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+describe CommentSerializer do
```
Comment (AlvaroSenorale) [inline spec/serializers/comment_serializer_spec.rb]: same comments than in ArticleSerializer

File: spec/rails_helper.rb (hunk 2 above)
Comment (MarcosFierro) [reply spec/rails_helper.rb]: that was trying to solve a configuration problem. It turned out that wasn't neccessary since Rails 4

File: spec/serializers/comment_serializer_spec.rb (hunk 5 above)
Comment (MarcosFierro) [reply spec/serializers/comment_serializer_spec.rb]: revised!

File: db/migrate/20191007205911_create_comments.rb [hunk 6]
```diff
@@ -1,8 +1,8 @@
 class CreateComments < ActiveRecord::Migration[5.2]
   def change
     create_table :comments do |t|
-      t.integer :user_id, :null => false
-      t.integer :article_id, :null => false
+      t.references :user_id, :null => false
```
Comment (guillezorron) [inline db/migrate/20191007205911_create_comments.rb]:
```suggestion
      t.references :user_id, null: false
```

File: db/migrate/20191007205911_create_comments.rb [hunk 7]
```diff
@@ -1,8 +1,8 @@
...
+      t.references :article_id, :null => false
```
Comment (guillezorron) [inline db/migrate/20191007205911_create_comments.rb]:
```suggestion
      t.references :article_id, null: false
```

File: spec/factories/comments.rb [hunk 8]
```diff
@@ -1,9 +1,10 @@
 # frozen_string_literal: true
+require 'faker'
 
 FactoryBot.define do
   factory :comment do
     user_id { 1 }
     article_id { 1 }
```
Comment (guillezorron) [inline spec/factories/comments.rb]: Here you can use `article_id { Article.first.id || FactoryBot.create(:article)` }

File: spec/factories/comments.rb [hunk 9]
```diff
@@ -1,9 +1,10 @@
 # frozen_string_literal: true
+require 'faker'
 
 FactoryBot.define do
   factory :comment do
     user_id { 1 }
```
Comment (guillezorron) [inline spec/factories/comments.rb]: Same below

File: spec/serializers/article_serializer_spec.rb [hunk 10]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
```
Comment (guillezorron) [inline spec/serializers/article_serializer_spec.rb]: Maybe you should do a easier test like:
```
let!(:user) { create(:user) }
let(:article) { create(:article) }
let(:serializer) { Api::V1::ArticleSerializer }
describe
  it
    expect(serializer.new(article).data.attributes.keys).to eql(%i[id type ...])
  end
end

```

Comment (guillezorron) [reply spec/serializers/article_serializer_spec.rb]: This example for all attributes and you have to do something similar with associations

File: spec/serializers/comment_serializer_spec.rb [hunk 11]
```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
```
Comment (guillezorron) [inline spec/serializers/comment_serializer_spec.rb]: Same above

**Comment by guillezorron** [comment]
Also we have to try to clarify better each PR, the name doesn't represent the work done in this PR.

File: spec/serializers/comment_serializer_spec.rb [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+      expect(CommentSerializer.new(comment).serialized_json).to eq(comment_attributes_json.to_json)
+    end
+  end
+
+  context 'another way to test' do
+    it 'checks hash data normal attributes' do
+      expect(serializer.new(comment).serializable_hash[:data][:attributes].keys).to eql(%i[id body])
+    end
+
+    it 'checks hash association\'s attributes' do
```
Comment (guillezorron) [inline spec/serializers/comment_serializer_spec.rb]:
```suggestion
    it 'checks hash associations attributes' do
```

File: spec/serializers/article_serializer_spec.rb [hunk 13]
```diff
@@ -1,24 +1,54 @@
...
+            "title": article.title,
+            "description": article.description },
+          "relationships":
+          { "comments":
+            { "data": [] },
+            "user":
+            { "data":
+              { "id": '1',
+                "type": 'user' } } } } }
+  end
```
Comment (guillezorron) [inline spec/serializers/article_serializer_spec.rb]: When you are testing serializers you have to check just the keys, because it's just the serializers work, the information of each key should be tested in controller test

Comment (guillezorron) [reply spec/serializers/article_serializer_spec.rb]: So you should remove this tests

File: spec/serializers/comment_serializer_spec.rb [hunk 14]
```diff
@@ -5,20 +5,50 @@
...
+        "relationships":
+        { "user":
+          { "data":
+            { "id": '1',
+              "type": 'user' } },
+          "article":
+          { "data":
+            { "id": '1',
+              "type": 'article' } } } } }
+  end
```
Comment (guillezorron) [inline spec/serializers/comment_serializer_spec.rb]: Same above
